workspaces/libnpmpublish/lib/provenance.js

```diff
@@ -1,69 +1,204 @@
 const { sigstore } = require('sigstore')
 const { readFile } = require('fs/promises')
+const ci = require('ci-info')
+const { env } = process
 
 const INTOTO_PAYLOAD_TYPE = 'application/vnd.in-toto+json'
 const INTOTO_STATEMENT_TYPE = 'https://in-toto.io/Statement/v0.1'
 const SLSA_PREDICATE_TYPE = 'https://slsa.dev/provenance/v0.2'
 
-const BUILDER_ID = 'https://github.com/actions/runner'
-const BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
-const BUILD_TYPE_VERSION = 'v2'
+const GITHUB_BUILDER_ID = 'https://github.com/actions/runner'
+const GITHUB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
+const GITHUB_BUILD_TYPE_VERSION = 'v2'
+
+const GITLAB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gitlab'
+const GITLAB_BUILD_TYPE_VERSION = 'v0alpha1'
```

Comment:
There was a question as to if we wanted this to change to beta?

Reply:
Also seems ok if you want to keep as is @wlynch?

Reply:
I'm fine with keeping this for now. I think there may be a few more tweaks coming to the provenance on the Fulcio side (https://github.com/sigstore/fulcio/pull/1206), but it shouldn't change anything w.r.t. npm. We can always rev this later.